DAOHaus + Harmony One Integration

victaphu opened the pull request:
This PR details the front-end PR to enable DAOHaus to support the Harmony One Blockchain.

victaphu committed:
updated: added harmony one configuration"

victaphu committed:
Merge branch 'HausDAO:develop' into develop

victaphu committed:
update: daohaus and harmony integration

skuhlmann commented on src/contexts/InjectedProviderContext.jsx:
```diff
@@ -36,6 +36,8 @@ export const InjectedProvider = ({ children }) => {
   const connectProvider = async () => {
     const providerOptions = getProviderOptions();
 
+    console.log(providerOptions);
```
can we remove this?

victaphu replied:
Yup let me remove

victaphu replied:
removed console.log

skuhlmann commented on src/utils/chain.js:
```diff
@@ -188,7 +188,7 @@ export const supportedChains = {
     abi_api_url:
       'https://api.polygonscan.com/api?module=contract&action=getabi&address=',
-    tokenlist_api_url: 'https://api.polygonscan.com/api',
+    tokenlist_api_url: 'https://api.harmony.one',
```
was updating the polygon api url a mistake?

victaphu replied:
yes that was a fat finger, i'll change it back

victaphu replied:
reverted back to the tokenlist api for polygon

skuhlmann commented on src/utils/chain.js:
```diff
@@ -221,6 +221,34 @@ export const supportedChains = {
+    boosts_graph_url:
+      'https://graph.t.hmny.io/subgraphs/name/harmony/daohaus-boost',
+    minion_factory_addr: '0x47373e8A3bEBb4920a64713AcA10aD9786eF8990',
+    moloch_factory_addr: '0x2880A45ad9121F6dB41463d81E598D7dD4D040cE',
```
can we get these contracts verified on the block explorer?

victaphu replied:
Verified following contracts:
MinionFactory https://explorer.harmony.one/verifycontract?address=0x47373e8a3bebb4920a64713aca10ad9786ef8990

Molochv2 https://explorer.harmony.one/verifycontract?address=0x2880a45ad9121f6db41463d81e598d7dd4d040ce

NiftyMinion https://explorer.harmony.one/verifycontract?address=0x29fa6df2514ddb8de516829c89b0caed84d74648

Pending ZAP contract (I'm working on that now). Can we proceed while this is in progress to remove blocker?

skuhlmann commented on src/utils/web3Modal.js:
```diff
@@ -10,6 +10,7 @@ export const attemptInjectedChainData = () => {
 
 const addNetworkProviders = chainData => {
   const allProviders = {};
+  console.log(chainData, window.ethereum?.chainId);
```
can we remove this?

skuhlmann commented on src/utils/exploreContent.js:
```diff
@@ -44,6 +44,18 @@ export const EXPLORE_FILTER_OPTIONS = [
+    default: true,
+  },
+  {
+    name: 'HarmonyTest',
```
Does the test network need to be in here? I don't see it in the chains.js.

victaphu replied:
Removed the testnet configuration from exploreContent

victaphu committed:
update: addressed pr comments from skuhlmann

removed console logs
reverted on polygon tokenlist url
removed testnet configuration

victaphu committed:
update: added support for safe minion config for harmony one

jordanlesich commented on src/utils/chain.js:
```diff
@@ -290,7 +323,10 @@ export const supportedChains = {
 
-export const chainByID = chainID => supportedChains[chainID];
+export const chainByID = chainID => {
+  console.log(chainID, supportedChains);
```
clean

victaphu replied:
Thanks, cleaned in latest release

jordanlesich reviewed (commented):
This all seems pretty straightforward. Not a whole lot to pick out.

victaphu committed:
update: cleaned up, remove console.log
